logic/src/main/java/com/grappus/logic/extensions/DatatypeExtensions.kt

```diff
@@ -0,0 +1,42 @@
+package com.grappus.logic.extensions
+
+/**
+ * Created by nimish@grappus.com on 16/12/2019.
+ * Description - Common data type extensions
+ */
+
+fun Float?.toFormattedString(): String {
+    return when {
+        this == null -> ""
+        this.toInt().toFloat() == this -> this.toInt().toString()
+        else -> this.toString()
+    }
+}
+
+fun Int?.toFormattedString(): String {
+    return when {
+        this == null -> ""
+        else -> this.toString()
+    }
+}
+
+fun Int?.orZero(): Int {
+    return when {
+        this == null -> 0
+        else -> this
+    }
+}
+
+fun Float?.orZero(): Float {
```

Comment:
Club `Float`, `Int`, `Long` & other data type functions together with label.

Reply:
Can't understand clubbing together. I have also written tests to check if they are working correctly.